File: PMUG02_16_2021_Project/Start_Java/app/src/main/java/com/neudesic/pmugfebappstart/AddPersonActivity.java
package com.neudesic.pmugfebappstart;

import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.neudesic.pmugfebappstart.viewmodel.PersonViewModel;

public class AddPersonActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_person);
        PersonViewModel viewModel = new ViewModelProvider(this, new ViewModelProvider.AndroidViewModelFactory(getApplication()) {
            @NonNull
            @Override
            public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
                PersonViewModel personViewModel = new PersonViewModel(getApplication());
                return (T)personViewModel;
            }
        }).get(PersonViewModel.class);

        EditText name = findViewById(R.id.name);
        EditText age = findViewById(R.id.age);
        EditText birthday = findViewById(R.id.birthday);

        Button addPerson = findViewById(R.id.action_add);
        addPerson.setOnClickListener(v -> {
            viewModel.addPerson(name.getText().toString(), Integer.parseInt(age.getText().toString()), birthday.getText().toString());
            Toast.makeText(getApplicationContext(), "Person added!", Toast.LENGTH_LONG).show();
        });
    }
}

File: PMUG02_16_2021_Project/End_Kotlin/app/src/main/java/com/neudesic/pmug02_16_2021_kotlin/AddPersonActivity.kt
package com.neudesic.pmug02_16_2021_kotlin

import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProvider.AndroidViewModelFactory
import com.neudesic.pmug02_16_2021_kotlin.viewmodel.PersonViewModel

class AddPersonActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_person)
        val viewModel: PersonViewModel = ViewModelProvider(this, object : AndroidViewModelFactory(
            application
        ) {
            override fun <T : ViewModel?> create(modelClass: Class<T>): T {
                val personViewModel = PersonViewModel(application)
                return personViewModel as T
            }
        }).get(PersonViewModel::class.java)
        val name = findViewById<EditText>(R.id.name)
        val age = findViewById<EditText>(R.id.age)
        val birthday = findViewById<EditText>(R.id.birthday)
        val addPerson = findViewById<Button>(R.id.action_add)
        addPerson.setOnClickListener { v: View? ->
            viewModel.addPerson(
                name.text.toString(),
                age.text.toString().toInt(),
                birthday.text.toString()
            )
            Toast.makeText(applicationContext, "Person added!", Toast.LENGTH_LONG).show()
        }
    }
}
